### PR TITLE
Ar 710 control the embed player from on an external site

### PR DESCRIPTION
**Pull request by guya**
video player api

### CODE REVIEW

File: public/api/video/kmediaPlayerApi.js [hunk 1]
```diff
@@ -0,0 +1,75 @@
+function KMediaPlayerApi(playerId, callbackFunc) {
+	this.playerId = playerId;
+	if (callbackFunc)
```
[the remark this answers is not in the document]
Comment (gshilin) [reply public/api/video/kmediaPlayerApi.js:3]: Only KMediaPlayerApi itself cannot be arrow function

File: src/components/AVPlayer/AVPlayer.js [hunk 2]
```diff
@@ -211,6 +212,79 @@ class AVPlayer extends Component {
...
+        break;
+      case 'mute':
+        media.mute(true);
+        break;
+      case 'muteUnmute':
+        media.muteUnmute();
+        break;
+      case 'setVolume':
+        if (event.data.volume === undefined) {
+          window.parent && window.parent.postMessage('playerCallback', '*');
```
[the remark this answers is not in the document]
Comment (guya) [reply src/components/AVPlayer/AVPlayer.js:242]: ok, iam fixing

File: src/components/AVPlayer/AVPlayer.js [hunk 3]
```diff
@@ -211,6 +212,79 @@ class AVPlayer extends Component {
...
+      case 'isPlaying':
+        this.sendCallbackMessage(event.data, { status: 'ok', result: media.isPlaying });
+        return;
+      default:
+        this.sendCallbackMessage(event.data, { status: 'error', error: 'command not found!' });
+        return;
+      }
+      this.sendCallbackMessage(event.data, { status: 'ok' });
+    } catch (e) {
+      console.error(e);
```
[the remark this answers is not in the document]
Comment (guya) [reply src/components/AVPlayer/AVPlayer.js:281]: ok, iam fixing

File: src/components/AVPlayer/AVPlayer.js [hunk 4]
```diff
@@ -211,6 +212,79 @@ class AVPlayer extends Component {
...
+      this.sendCallbackMessage(event.data, { status: 'error', error: e });
+    }
+  }
+
+  sendCallbackMessage(params, result) {
+    window.parent && window.parent.postMessage({params, result}, '*');
```
[the remark this answers is not in the document]
Comment (guya) [reply src/components/AVPlayer/AVPlayer.js:287]: ok

File: public/api/video/kmediaPlayerApi.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (guya) [reply public/api/video/kmediaPlayerApi.js:3]: ok

File: public/api/video/kmediaPlayerApi.js [hunk 5]
```diff
@@ -0,0 +1,75 @@
...
+		window.addEventListener('message', callbackFunc, false);
+
+	this.setPlayer = function(args) {
+		let player = document.getElementById(this.playerId).contentWindow;
```
[the remark this answers is not in the document]
Comment (guya) [reply public/api/video/kmediaPlayerApi.js:7]: ok, i will add error check + throw

File: src/components/AVPlayer/AVPlayer.js [hunk 6]
```diff
@@ -211,6 +212,79 @@ class AVPlayer extends Component {
       clearTimeout(this.autohideTimeoutId);
       this.autohideTimeoutId = null;
     }
+    window.removeEventListener('message');
```
[the remark this answers is not in the document]
Comment (guya) [reply src/components/AVPlayer/AVPlayer.js:215]: ok

**Pull request merged**